events/syscall/unexpected_k8s_nodeport_connection.go

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+Copyright (C) 2024 The Falco Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package syscall
+
+import (
+	"errors"
+	"github.com/falcosecurity/event-generator/events"
+	"net"
+	"os/exec"
+	"strconv"
+)
+
+var _ = events.Register(
+	UnexpectedK8sNodePortConnection,
+	events.WithDisabled(), // this rule is not included in falco_rules.yaml (stable rules), so disable the action
+)
+
+func UnexpectedK8sNodePortConnection(h events.Helper) error {
+	if h.InContainer() {
+		port := 31000
+
+		hostIP, err := getHostEth0IP()
+		if err != nil {
+			return err
+		}
+
+		cmd := exec.Command("nc", hostIP, strconv.Itoa(port), "<", "/dev/null")
+		err = cmd.Run()
+		if err != nil {
+			return err
+		}
+	}
+
+	return &events.ErrSkipped{
+		Reason: "'Unexpected k8s Nodeport connection' is applicable only to containers.",
+	}
+}
+
+func getHostEth0IP() (string, error) {
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		return "", err
+	}
+
+	for _, iface := range ifaces {
+		if iface.Name == "eth0" {
+			addrs, err := iface.Addrs()
+			if err != nil {
+				return "", err
+			}
+
+			for _, addr := range addrs {
+				ipnet, ok := addr.(*net.IPNet)
+				if ok && !ipnet.IP.IsLoopback() {
+					if ipnet.IP.To4() != nil {
+						return ipnet.IP.String(), nil
+					}
+				}
+			}
+		}
+	}
+
+	return "", errors.New("eth0 interface not found or has no valid IPv4 address")
+}
```

Comment:
Could we move this function to another file (i.e. `utils_net.go`)?

This will improve reusability.

Reply:
Yes @leogr I shifted `getHosteth0IP` function to utils_net.go under `events/syscall`